lib/utils/fncache.go

```diff
@@ -111,14 +117,70 @@ type fnCacheEntry struct {
 	v      any
 	e      error
 	t      time.Time
+	ttl    time.Duration
 	loaded chan struct{}
 }
 
+// Shutdown expires all items in the cache. If the OnExpires
+// callback was set in the FnCacheConfig it will be called once
+// per item in the cache.
+func (c *FnCache) Shutdown(ctx context.Context) {
+	c.mu.Lock()
+	c.closed = true
+	entries := c.entries
+	c.entries = make(map[any]*fnCacheEntry)
+	c.mu.Unlock()
```

Comment:
nit: I think we probably want to cancel in-flight load operations at the start of the shutdown operation (but after we acquire the lock, so that it does not also cancel any inflight evictions).  Loads are currently scoped directly under the parent context passed in via config, so we'd need to setup in intermediate context in `NewFnCache` and replace all our invocations of `c.cfg.Context` with that.